userguide/sound_classifier/how-it-works.md

```diff
@@ -0,0 +1,52 @@
+# How Does this Work?
+
+Training and making predctions, for a sound classifier model, is a three
+stage process. The first stage does signal preprocessing. In
+the second step we use a pretrained neural network to extract deep
+features. Finally in the last step, we use a custom neural network.
+
+## Signal Preprocessing Pipeline Stage
+Several audio transformation happen during this stage. For someone without
+sound domain knowledge, this is probably the most complicated part.
+Nothing about this stage is updated based on the input data (i.e. nothing
+is learned in this stage).
+
+At a high level the preprocessing pipeline does the following:
+* The raw pulse code modulation data from the wav file is converted to
+floats on a [-1.0, +1.0] scale.
+* The data is resampled to only 16,000 sample per second.
+* The data is broken up in to several overlapping windows.
+* A [Hamming Window](https://en.wikipedia.org/wiki/Window_function#Hann_and_Hamming_windows) is applied to each windows.
+* The [Power Spectrum](https://en.wikipedia.org/wiki/Spectral_density#Power_spectral_density) is calculated, using a [Fast Fourier Transformation](https://en.wikipedia.org/wiki/Fourier_transform).
+* Frequencies above and below certain thresholds are dropped.
+* Mel Frequency Filter Banks are applied.
+* Finally the log is taken of all values.
+
+The preprocessing pipeline takes 975ms worth of audio as input (exact
+input lenght depends on sample rate) and produce an array of shape
+(96, 64).
+
+## VGGish Feature Extraction Stage
+VGGish is a pretrained [Convolutional Neural Network](https://en.wikipedia.org/wiki/Convolutional_neural_network) from Google,
+see [their paper](https://ai.google/research/pubs/pub45611) and [their GitHub page](https://github.com/tensorflow/models/tree/master/research/audioset) for more details. As the name suggests, the architecture of
+this network is inspired by the famous VGG networks used for image
+classification. The network consists of a series of convolution and
+activation layers, optionally followed by a max pooling layer.
+This network contains 17 layers in total.
+
+This network is kept static during model training. We have removed the
+last three layers of the original VGGish model. We use the widest
+layer, from the original network, as our input data for the final
```

Comment:
Two general questions: 
1. Is the widest layer from the original network the same as the fourth last layer in the original network? 
2. Are the three layers we have in our custom neural network the same as the last three layers in VGGish?

Reply:
1 - Yes, the widest layer is the fourth to the last layer in the original VGGish network.
2 - No, those have significantly more (4096) nodes. In prototyping testing I did fewer (100) nodes actually preformed better. It is also of course faster to train and predict too.